### PR TITLE
[Fix][Connector-V2][FTP] Fix FTP connector connection_mode is not effective

### LINKED ISSUE #7863: [Bug] [Connector-V2][FTP] FTP connector connection_mode is not effective
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Configuring connection_mode does not take effect
```
source {
  FtpFile {
    host= "ip"
    port= 21
    connection_mode= "passive_local"
    user= "admin"
    password= "admin"
    path= "/123.txt"
    file_format_type= "binary"
    encoding = "UTF-8"
  }
}
```
Exception:
image

Cause of the problem:
Described in the document: You can set active_local and passive_local, 
that is: connection_mode = "active_local"
image

In `FtpConnectionMode.java` you can see the implementation code:
image

But when ConfigUtil parses the mode, it tries to filter by ACTIVE_LOCAL_DATA_CONNECTION_MODE = active_local, which is wrong.

image

This will eventually cause the enumeration parsing to fail and return an exception.
image

Resolve method:
The name in the enumeration must be consistent with the logic of the ConfigUtil.convertToEnum method

image

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
source {
  FtpFile {
    host= "ip"
    port= 21
    connection_mode= "passive_local"
    user= "admin"
    password= "admin"
    path= "/123.txt"
    file_format_type= "binary"
    encoding = "UTF-8"
  }
}
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangshenghang**
### Purpose of this pull request

- reslove #7863

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-ftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/ftp/system/SeaTunnelFTPFileSystem.java [hunk 1]
```diff
@@ -172,13 +169,14 @@ private FTPClient connect() throws IOException {
...
-            case ACTIVE_LOCAL_DATA_CONNECTION_MODE:
+            case ACTIVE_LOCAL:
                 client.enterLocalActiveMode();
                 break;
-            case PASSIVE_LOCAL_DATA_CONNECTION_MODE:
+            case PASSIVE_LOCAL:
                 client.enterLocalPassiveMode();
                 break;
             default:
+                System.out.println("Unsupported FTP connection mode: " + mode);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-ftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/ftp/system/SeaTunnelFTPFileSystem.java]: please use log.

**Review by Hisoka-X — COMMENTED** [review]
please add test case

**Comment by zhangshenghang** [comment]
> please add test case

I encountered a problem. Docker runs ftp in passive mode through the following command
```bash
docker run -d \
--name ftp-server \
-p 20:20 \
-p 21:21 \
-p 990:990 \
-p 30000-30099:30000-30099 \
-e FTP_USER=admin \
-e FTP_PASS=admin \
-e PASV_ADDRESS=172.16.25.204 \
-e PASV_MIN_PORT=30000 \
-e PASV_MAX_PORT=30099 \
-v /data/data:/home/vsftpd \
fauria/vsftpd
```
Need to specify the `30000-30099:30000-30099` port range to reserve for passive mode.

I tried many configuration methods through `private GenericContainer<?> ftpContainer;`, but it doesn't seem to support starting with reserved ports. You have to wait for all open ports to run successfully before ftpContainer can continue to run.

image

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @zhangshenghang !

**Review by hailin0 — APPROVED** [review]

**Pull request merged**